external-crates/move/crates/move-package/src/source_package/manifest_parser.rs

```diff
@@ -535,11 +557,20 @@ fn parse_dep_override(tval: TV) -> Result<PM::DepOverride> {
     Ok(tval.as_bool().unwrap())
 }
 
-// check that only recognized names are provided at the top-level
-fn warn_if_unknown_field_names(table: &toml::map::Map<String, TV>, known_names: &[&str]) {
+// Check that only recognized names, or tables prefixed by a known prefix are provided at the
+// top-level.
+fn warn_if_unknown_field_names(
+    table: &toml::map::Map<String, TV>,
+    known_names: &[&str],
+    known_prefixes: &[&str],
+) {
     let mut unknown_names = BTreeSet::new();
     for key in table.keys() {
-        if !known_names.contains(&key.as_str()) {
+        if !known_names.contains(&key.as_str())
```

Comment:
I kind of expected the resolver name to show up as a key called `r` here, meaning you wouldn't need to handle prefixes specially -- was that not the case?

Reply:
Just hadn't thought about/realized that. Can absolutely just look for a key called `r`.